packages/service-clients/end-to-end-tests/azure-client/src/test/presenceTest.spec.ts

```diff
@@ -0,0 +1,175 @@
+
+interface MessageFromChild {
+	event: string;
+	sessionId: string;
+}
+
+interface MessageToParent {
+	command: string;
+	containerId: string;
+	user: AzureUser;
+}
+
+describe(`Presence with AzureClient`, () => {
+	const numClients = 5;
+	const children: ChildProcess[] = [];
+	const connectTimeoutMs = 10_000;
+	const initialUser: AzureUser = {
+		id: "test-user-id-1",
+		name: "test-user-name-1",
+	};
+
+	afterEach(async () => {
+		// kill all child processes after each test
+		for (const child of children) {
+			if (!child.killed) {
+				child.kill();
+			}
+		}
+	});
+
+	const createPresenceContainer = async (
+		user: AzureUser,
+		config?: ReturnType<typeof configProvider>,
+		scopes?: ScopeType[],
+	): Promise<{
+		container: IFluidContainer;
+		presence: IPresence;
+		services: AzureContainerServices;
+		client: AzureClient;
+		containerId: string;
+	}> => {
+		const client = createAzureClient(user.id, user.name, undefined, config, scopes);
+		const schema: ContainerSchema = {
+			initialObjects: {
+				presence: ExperimentalPresenceManager,
+			},
+		};
+
+		const { container, services } = await client.createContainer(schema, "2");
+		const containerId = await container.attach();
+
+		if (container.connectionState !== ConnectionState.Connected) {
+			await timeoutPromise((resolve) => container.once("connected", () => resolve()), {
+				durationMs: connectTimeoutMs,
+				errorMsg: "container connect() timeout",
+			});
+		}
+
+		assert.strictEqual(typeof containerId, "string", "Attach did not return a string ID");
+		assert.strictEqual(
+			container.attachState,
+			AttachState.Attached,
+			"Container is not attached after attach is called",
+		);
+
+		const presence = acquirePresenceViaDataObject(
+			container.initialObjects.presence as ExperimentalPresenceDO,
+		);
+		return {
+			client,
+			container,
+			presence,
+			services,
+			containerId,
+		};
+	};
+
+	it("announces 'attendeeDisconnected' when remote client disconnects", async () => {
+		const attendeesJoined: string[] = [];
+		const attendeesDisconnected: string[] = [];
+
+		// Parent proceess creates container
+		const { container, presence, containerId } = await createPresenceContainer(initialUser);
+
+		// Fork child processes
+		for (let i = 0; i < numClients; i++) {
+			const user = { id: `test-user-id-${i}`, name: `test-user-name-${i}` };
+			const child = fork("./lib/test/childClient.js");
+			children.push(child);
+
+			// Send connect command to child
+			const message: MessageToParent = { command: "connect", containerId, user };
+			child.send(message);
+		}
+
+		// Wait to receive attendeeJoined event on presence from remote attendees
+		await timeoutPromise(
+			(resolve) => {
+				let count = 0;
+				presence.events.on("attendeeJoined", (attendee) => {
+					// Only account for remote attendees
+					if (attendee !== presence.getMyself()) {
+						count++;
+						attendeesJoined.push(attendee.sessionId);
+						if (count === numClients) {
```

Comment:
nit: You could get rid of `count` by comparing `attendeesJoined.length === numClients` instead